tachyon/math/polynomials/univariate/radix2_evaluation_domain.h

```diff
@@ -440,6 +447,31 @@ class Radix2EvaluationDomain
     }
   }
 
+  // Expands a |Eigen::MatrixBase|'s rows from |rows| to |rows|^(|added_bits|),
+  // moving values from row |i| to row |i|^(|added_bits|). All new entries are
+  // set to |F::Zero()|. Returns false when |added_bits| is zero, and no
+  // expansion is performed.
+  template <typename Derived, typename Derived2>
+  [[nodiscard]] CONSTEXPR_IF_NOT_OPENMP static bool ExpandWithZeroPad(
+      Eigen::MatrixBase<Derived>& mat, size_t added_bits,
+      Eigen::MatrixBase<Derived2>& out) {
+    if (added_bits == 0) return false;
```

Comment:
The return value of `ExpandWithZeroPad()` seems redundant since the only place it is used already checks if `added_bits` is 0. Could you remove the return type?